### PR TITLE
Use SafeLoggable.getLogMessage() when available during rewrap

### PR DESCRIPTION
**Pull request by henryptung**
Don't risk leaking unsafe information from getMessage() into a
SafeLoggable constructor (which is likely declared @CompileTimeConstant).

### CODE REVIEW

**Comment by pustaczek** [comment]
This code path seemingly doesn't end up being hit in the AtlasDB codebase, because if [constructors don't fall into this pattern](https://github.com/palantir/atlasdb/blob/d35ec651cd6150e814661e027285bc908b63b837/atlasdb-commons/src/main/java/com/palantir/common/base/Throwables.java#L207-L215), the rewrap gives up and returns the original exception. No safe exceptions I could find have such constructors, though the change would still make sense just out of caution. Is this intended?

**Comment by henryptung** [comment]
Yep, mostly just a safety fix for downstream consumers using the Throwables lib (and especially if they implement their own SafeLoggable exceptions, which may hit the relevant constructor signatures).

File: atlasdb-commons/src/test/java/com/palantir/common/base/ThrowablesTest.java [hunk 1]
```diff
@@ -90,6 +98,25 @@ public void testThrowCauseAsUnchecked() {
...
+        assertThat(rewrapped1.getLogMessage()).doesNotContain("secret");
+
+        SafeRewrappableException rewrapped2 = null;
+        try {
+            Throwables.rewrapAndThrowIfInstance(rewrapped1, SafeRewrappableException.class);
+        } catch (SafeRewrappableException ex) {
+            rewrapped2 = ex;
+        }
+        assertThat(rewrapped2).isNotNull();
+        assertThat(rewrapped2.getLogMessage()).doesNotContain("secret");
```
Comment (jeremyk-91) [inline atlasdb-commons/src/test/java/com/palantir/common/base/ThrowablesTest.java:117]: For this section, it probably suffices to use `assertThatThrownBy(...)`?

**Review by jeremyk-91 — COMMENTED** [review]
Thanks for adding the test. I think it can be written a bit more simply/clearly, but otherwise looks good.

**Review by jeremyk-91 — APPROVED** [review]
👍

**Pull request merged**